genesyscloud/idp_adfs/resource_genesyscloud_idp_adfs.go

```diff
@@ -0,0 +1,145 @@
+}
+
+// readIdpAdfs is used by the idp_adfs resource to read an idp adfs from genesys cloud
+func readIdpAdfs(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getIdpAdfsProxy(sdkConfig)
+
+	log.Printf("Reading idp adfs %s", d.Id())
+
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceIdpAdfs(), constants.DefaultConsistencyChecks, resourceName)
+
+	return util.WithRetriesForReadCustomTimeout(ctx, d.Timeout(schema.TimeoutRead), d, func() *retry.RetryError {
+		aDFS, resp, getErr := proxy.getIdpAdfs(ctx)
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				createIdpAdfs(ctx, d, meta)
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read IDP ADFS: %s", getErr), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read IDP ADFS: %s", getErr), resp))
+		}
+
+		resourcedata.SetNillableValue(d, "disabled", aDFS.Disabled)
+		resourcedata.SetNillableValue(d, "issuer_uri", aDFS.IssuerURI)
+		resourcedata.SetNillableValue(d, "target_uri", aDFS.SsoTargetURI)
+		resourcedata.SetNillableValue(d, "relying_party_identifier", aDFS.RelyingPartyIdentifier)
+
+		if aDFS.Certificate != nil {
+			d.Set("certificates", lists.StringListToInterfaceList([]string{*aDFS.Certificate}))
+		} else if aDFS.Certificates != nil {
+			d.Set("certificates", lists.StringListToInterfaceList(*aDFS.Certificates))
+		} else {
+			d.Set("certificates", nil)
+		}
+
+		log.Printf("Read idp adfs")
+		return cc.CheckState(d)
+	})
+}
+
+// updateIdpAdfs is used by the idp_adfs resource to update an idp adfs in Genesys Cloud
+func updateIdpAdfs(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getIdpAdfsProxy(sdkConfig)
+
+	idpAdfs := getIdpAdfsFromResourceData(d)
+	certificates := lists.BuildSdkStringListFromInterfaceArray(d, "certificates")
+	if certificates != nil {
+		if len(*certificates) == 1 {
+			idpAdfs.Certificate = &(*certificates)[0]
+		}
+		idpAdfs.Certificates = certificates
+	}
+	log.Printf("Updating idp adfs")
+	resp, err := proxy.updateIdpAdfs(ctx, d.Id(), &idpAdfs)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update IDP ADFS %s error: %s", d.Id(), err), resp)
+	}
+
+	log.Printf("Updated idp adfs")
+	return readIdpAdfs(ctx, d, meta)
+}
+
+// deleteIdpAdfs is used by the idp_adfs resource to delete an idp adfs from Genesys cloud
+func deleteIdpAdfs(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getIdpAdfsProxy(sdkConfig)
+
+	_, err := proxy.deleteIdpAdfs(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to delete idp adfs %s: %s", d.Id(), err)
+	}
```

Comment:
You can change this to return the `BuildAPIDiagnosticError` function. You'll need to replace `_` with `resp` to catch the API Response object